### PR TITLE
feat: add xapi transformer for grading events

### PR DESCRIPTION
**Pull request by Ian2012**
**Description:** Adds transformers for `edx.grades.subsection.grade_calculated` and `edx.grades.course.grade_calculated` events.

**JIRA:** (https://github.com/openedx/event-routing-backends/issues/331)

**Dependencies:** dependencies on other outstanding PRs, issues, etc.

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

**Comment by Ian2012** [comment]
@ziafazal what do you think about the proposed schema? Do you think we need to change it?

**Comment by ziafazal** [comment]
> @ziafazal what do you think about the proposed schema? Do you think we need to change it?

@Ian2012 I think we need to change schema in these events too. We should add `result` in xAPI statements with grades
```
  "result": {
        "score": {
            "max": 1,
            "min": 0,
            "raw": 0,
            "scaled": 0
        },
        "success": false
    },
```
Result can be calculated same as [here](https://github.com/openedx/event-routing-backends/blob/143e821d903b2be999ddebe51ca6e11b07b133f6/event_routing_backends/processors/xapi/event_transformers/problem_interaction_events.py#L146)

File: event_routing_backends/processors/xapi/tests/fixtures/expected/edx.grades.course.grade_calculated.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "id": "32e08e30-f8ae-4ce2-94a8-c2bfe38a70cb",
+  "result": {
+    "score": {
+      "scaled": 0.02,
+      "raw": 0.02,
+      "min": 0.0,
+      "max": 1.0
+    },
+    "extensions": {}
```
Comment (bmtcril) [inline event_routing_backends/processors/xapi/tests/fixtures/expected/edx.grades.course.grade_calculated.json:10]: I think this should just be omitted if it's empty?

**Review by bmtcril — COMMENTED** [review]
Just one comment, otherwise lgtm

File: event_routing_backends/processors/xapi/tests/fixtures/expected/edx.grades.course.grade_calculated.json (hunk 1 above)
Comment (Ian2012) [reply event_routing_backends/processors/xapi/tests/fixtures/expected/edx.grades.course.grade_calculated.json]: It was fixed by setting a default value for empty `letter_grade`.

**Review by bmtcril — APPROVED** [review]
LGTM

**Comment by Ian2012** [comment]
cc @ziafazal

**Review by ziafazal — APPROVED** [review]
👍

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/event-routing-backends&entry.867055334=https://github.com/openedx/event-routing-backends/pull/341&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-08-23+19:00&entry.1917517419=2023-08-29+12:52&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**